cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/adyen_pos.js

```diff
@@ -0,0 +1,91 @@
+/**
+ *
+ */
+
+'use strict';
+
+var server = require("server");
+var collections = require("*/cartridge/scripts/util/collections");
+var Resource = require("dw/web/Resource");
+var Transaction = require("dw/system/Transaction");
+var Order = require('dw/order/Order');
+var Logger = require('dw/system/Logger');
+
+function Handle(basket, paymentInformation) {
+    Transaction.wrap(function () {
+        collections.forEach(basket.getPaymentInstruments(), function (item) {
+            basket.removePaymentInstrument(item);
+        });
+
+        var paymentInstrument = basket.createPaymentInstrument(
+            "AdyenPOS", basket.totalGrossPrice
+        );
+        paymentInstrument.custom.adyenPaymentMethod = "POS Terminal";
+    });
+
+    return {error: false};
+}
+
+/**
+ * Authorize
+ */
+function Authorize(orderNumber, paymentInstrument, paymentProcessor) {
+    var adyenTerminalApi = require("*/cartridge/scripts/adyenTerminalApi");
+    Transaction.wrap(function () {
+        paymentInstrument.paymentTransaction.transactionID = orderNumber;
+        paymentInstrument.paymentTransaction.paymentProcessor = paymentProcessor;
+    });
+
+    var adyenPaymentForm = server.forms.getForm("billing").adyenPaymentFields;
+    var OrderMgr = require("dw/order/OrderMgr");
+    var order = OrderMgr.getOrder(orderNumber);
+
+    var terminalId = null;
+    if(adyenPaymentForm.terminalId.value){
+        terminalId = adyenPaymentForm.terminalId.value;
+    }
```

Comment:
should we not do the oposit. So  if(!adyenPaymentForm.terminalId.value){ break here and throw error as terminalId is empty } ?